### ISSUE TITLE
Sticky Broken Thumbnail

### ISSUE THREAD

**Issue by garrettwrong** [sphinx-gallery/sphinx-gallery#1051]
Hi, first thanks for `sphinx-gallery`, it has been very useful to our project. I'd like to report an issue we've run into with the gallery thumbnails.  

When we are developing the gallery, sometimes it will fail. After we fix the gallery, the thumbnails still show "BROKEN". I would expect them to be updated when successful.

We have been working around this by running a clean command to remove the `auto_*` directories and regenerating, but this can take some time for our examples.

Thanks!

**Comment by lucyleeow** [comment]
Sorry I don't have a lot of capacity but I can try to take a look if I get a chance.

@garrettwrong - just to confirm, after the gallery example is fixed, the example is updated but the thumbnail does not update from 'broken'? But forcing all examples to be re-run works?

**Comment by garrettwrong** [comment]
Yes. When this occurs we remove that directory and rebuild the entire gallery.  Thanks!
